File: hcx-pipeline-jobs/jobs-core/src/main/java/org/swasth/job/connector/FlinkKafkaConnector.java
package org.swasth.job.connector;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer;
import org.apache.flink.streaming.connectors.kafka.KafkaSerializationSchema;
import org.swasth.job.Platform;

import java.util.Properties;

public class FlinkKafkaConnector {

    private final String kafkaBrokerServers = Platform.getString("bootstrap-servers","localhost:9092");

    public FlinkKafkaConsumer<String> kafkaStringSource(String kafkaTopic) {
        return new FlinkKafkaConsumer<String>(kafkaTopic, new SimpleStringSchema(), kafkaConsumerProperties());
    }

    public FlinkKafkaProducer<String> kafkaStringSink(String kafkaTopic) {
        return new FlinkKafkaProducer<String>(kafkaTopic, new SimpleStringSchema(), kafkaProducerProperties());
    }

    public Properties kafkaConsumerProperties(){
        Properties properties= new Properties();
        properties.put("bootstrap.servers", kafkaBrokerServers);
        return properties;
    }

    public Properties kafkaProducerProperties(){
        Properties properties= new Properties();
        properties.put("bootstrap.servers", kafkaBrokerServers);
        return properties;
    }

}

File: hcx-pipeline-jobs/core/src/main/scala/org/swasth/dp/core/job/FlinkKafkaConnector.scala
package org.swasth.dp.core.job

import org.apache.flink.connector.base.DeliveryGuarantee
import org.apache.flink.connector.kafka.sink.KafkaSink
import org.apache.flink.connector.kafka.source.KafkaSource
import org.swasth.dp.core.serde._

import java.util._
class FlinkKafkaConnector(config: BaseJobConfig) extends Serializable {

def kafkaMapSource(kafkaTopic: String): KafkaSource[java.util.Map[String, AnyRef]] = {
  KafkaSource.builder[java.util.Map[String, AnyRef]]()
    .setTopics(kafkaTopic)
    .setDeserializer(new MapDeserializationSchema)
    .setProperties(config.kafkaConsumerProperties)
    .build()
  }

  def kafkaMapSink(kafkaTopic: String): KafkaSink[java.util.Map[String, AnyRef]] = {
    KafkaSink.builder[java.util.Map[String, AnyRef]]()
      .setDeliveryGuarantee(DeliveryGuarantee.AT_LEAST_ONCE)
      .setRecordSerializer(new MapSerializationSchema(kafkaTopic))
      .setKafkaProducerConfig(config.kafkaProducerProperties)
      .build()
  }

  def kafkaStringSource(kafkaTopic: String): KafkaSource[String] = {
    KafkaSource.builder[String]()
      .setTopics(kafkaTopic)
      .setDeserializer(new StringDeserializationSchema)
      .setProperties(config.kafkaConsumerProperties)
      .build()
  }

  def kafkaStringSink(kafkaTopic: String): KafkaSink[String] = {
    KafkaSink.builder[String]()
      .setDeliveryGuarantee(DeliveryGuarantee.AT_LEAST_ONCE)
      .setRecordSerializer(new StringSerializationSchema(kafkaTopic))
      .setKafkaProducerConfig(config.kafkaProducerProperties)
      .build()
  }
}